community/dbms/src/main/java/org/neo4j/server/configuration/ConfigLoader.java

```diff
@@ -21,53 +21,62 @@
 
 import java.io.File;
 import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import org.neo4j.graphdb.factory.GraphDatabaseSettings;
 import org.neo4j.helpers.collection.Pair;
 import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.configuration.ConfigurationValidator;
+import org.neo4j.kernel.configuration.Connector;
 import org.neo4j.kernel.configuration.Settings;
 
 public class ConfigLoader
 {
     public static final String DEFAULT_CONFIG_FILE_NAME = "neo4j.conf";
 
-    private final Function<Map<String, String> ,Iterable<Class<?>>> settingsClassesSupplier;
+    public ConfigLoader()
+    {
+    }
```

Comment:
Why do we need a public constructor as well as the static factory methods? If we can't hid the constructor for some reason then this empty no arg public constructor is not necessary.